File: 조준장/level3/이중우선순위큐.py
import heapq

def solution(operations):
    heap = []
    
    for operation in operations:
        o, num = operation.split()
        num = int(num)
        
        if o == "I":
            heapq.heappush(heap, num)
        
        elif heap:
            if num == -1:
                heapq.heappop(heap)
            else:
                heap.remove(max(heap))
                heapq.heapify(heap)
                
    if not len(heap):
        return [0, 0]

    return [max(heap), heap[0]]

File: 조준장/level3/이중우선순위큐.kt
class Solution {
    fun solution(operations: Array<String>): IntArray {
        var answer = mutableListOf<Int>()
        operations
            .map{it.split(" ")}
            .map{Pair(it[0], it[1].toInt())}
            .forEach{ 
                when (it.first){
                    "I" -> answer.add(it.second)
                    "D" -> { 
                        if (!answer.isEmpty()){
                            val num : Int = 
                            if (it.second == 1) { answer.maxOf{it} }
                            else { answer.minOf{it}}
                            answer.remove(num)
                        }
                    }
                }
            }
            
           return if (answer.isEmpty()) intArrayOf(0, 0)
           else intArrayOf(answer.maxOf{it}, answer.minOf{it})
    }
}
